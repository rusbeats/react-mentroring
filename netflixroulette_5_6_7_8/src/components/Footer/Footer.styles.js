import constants from '../../utils/constants';
import styled from 'styled-components';

const { colors } = constants;

export const StyledFooter = styled.footer`
    padding: 20px 0 20px 75px;
    background-color: ${colors.lightBlack};
    color: ${colors.lightPink};
    font-weight: bold;
`;
