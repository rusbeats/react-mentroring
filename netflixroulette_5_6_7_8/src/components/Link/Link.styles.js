import styled from 'styled-components';

// Implemented link
export const StyledLink = styled.a`
    text-decoration: none;
    color: ${props => props.color};
    cursor: pointer;

    &:focus,
    &:hover,
    &:visited,
    &:link,
    &:active {
        text-decoration: none;
    }
`;
