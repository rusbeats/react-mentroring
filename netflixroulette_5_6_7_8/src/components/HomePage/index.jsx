import React, { Fragment } from 'react';
import Header from '../Header';
import Films from '../FilmsList/Films';
import Footer from '../Footer';

//Wrapped in Fradment
//Removed all unnecessary divs
//included Header and Footer
const HomePage = () => {
    return (
        <Fragment>
            <Header />
            <Films />
            <Footer />
        </Fragment>
    );
};

export default HomePage;
