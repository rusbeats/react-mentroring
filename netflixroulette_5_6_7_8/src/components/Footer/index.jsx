import React from 'react';
import { useTranslation } from 'react-i18next';
import { StyledLink } from '../Link/Link.styles';
import { StyledFooter } from './Footer.styles';
import constants from '../../utils/constants';
import Link from 'next/link';

const { colors } = constants;

const Footer = () => {
    const { t } = useTranslation();
    return (
        <StyledFooter>
            <Link href="/">
                <StyledLink color={colors.lightPink}>{t('netflixRoulette')}</StyledLink>
            </Link>
        </StyledFooter>
    );
};

StyledFooter.displayName = 'Footer';

export default Footer;
