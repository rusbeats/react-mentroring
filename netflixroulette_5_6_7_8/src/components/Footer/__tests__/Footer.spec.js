import React from 'react';
import { shallow } from 'enzyme';
import Footer from '../index';
import { StyledLink } from '../../Link/Link.styles';
import HomePage from '../../HomePage';

describe('Footer', () => {
    it('should render correctly (snapshot)', () => {
        const component = shallow(<Footer />);
        expect(component).toMatchSnapshot();
    });

    it('should go to Home', () => {
        const component = shallow(<Footer />);
        const homePage = shallow(<HomePage />);
        component.find(StyledLink).simulate('click');
        expect(homePage).toHaveLength(1);
    });
});
