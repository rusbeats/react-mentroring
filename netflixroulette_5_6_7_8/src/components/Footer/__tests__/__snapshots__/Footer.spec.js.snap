// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render correctly (snapshot) 1`] = `
<Footer>
  <Link
    href="/"
  >
    <Linkstyles__StyledLink
      color="#d15c6e"
    >
      netflixRoulette
    </Linkstyles__StyledLink>
  </Link>
</Footer>
`;
